src/librbd/AioImageRequest.cc

```diff
@@ -382,6 +390,107 @@ void AioImageWrite::update_stats(size_t length) {
   m_image_ctx.perfcounter->inc(l_librbd_wr_bytes, length);
 }
 
+void AioImageWriteSame::assemble_extent(const ObjectExtent &object_extent,
+                                    bufferlist *bl) {
+  if (object_extent.offset % m_data_len == 0
```

Comment:
hmm..i mistake, this should be (object_extent.length % m_data_len == 0), and should make a judgement that striper_units is a multiple of data_len before assemble_extent, otherwise, the current method cannot handle。
should we handle the situation striper_units is not a multiple of data_len, it seems too complex, in block device application scenario, this could never happen？